src/lib/prov/tpm2/tpm2_key.h

```diff
@@ -35,6 +35,16 @@ namespace Botan::TPM2 {
  * @param public_blob The public blob to load as an ordinary RSA key
  */
 BOTAN_PUBLIC_API(3, 6) Botan::RSA_PublicKey rsa_pubkey_from_tss2_public(const TPM2B_PUBLIC* public_blob);
+
+/**
+ * This helper function transforms a @p public_blob in a TPM2B_PUBLIC* format
+ * into the functional components of an RSA public key. Namely, a pair of
+ * modulus and exponent as big integers.
+ *
+ * @param public_blob The public blob to decompose into RSA pubkey components
+ */
+BOTAN_PUBLIC_API(3, 7)
+std::pair<BigInt, BigInt> rsa_pubkey_components_from_tss2_public(const TPM2B_PUBLIC* public_blob);
```

Comment:
Do we really need this as a public function? It seems like external callers could use `rsa_pubkey_from_tss2_public` as usual

Reply:
I explicitly published it mostly for consistency with the ECC variant that also returns the public key as a `std::pair<>` of its components. No strong feeling about it either way.

Reply:
I moved the "components" variant into `tpm2_rsa.h` and removed the `BOTAN_PUBLIC_API`.